pkg/kn/flags/bool.go

```diff
@@ -65,6 +68,25 @@ func ReconcileBoolFlags(f *pflag.FlagSet) error {
 		if err != nil {
 			return
 		}
+
+		// handle DEPRECATD flags
+		if strings.HasPrefix(flag.Usage, deprecatedPrefix) {
+			if flag.Changed {
+				// handle async flag
+				if flag.Name == "async" {
+					if f.Lookup("wait").Changed || f.Lookup("no-wait").Changed {
+						err = fmt.Errorf("only one of (DEPRECATED) --%s, --wait and --no-wait may be specified", flag.Name)
```

Comment:
You already know that the flag name is "async", no need to add it via parameter here.